### ISSUE TITLE
Only officially support node 8 and higher from Commander v4

### ISSUE THREAD

**Issue by shadowspawn** [tj/commander.js#1045]
We have been planning for a long time to use more modern javascript features, but changes will break older versions of node and we do not have a documented policy.

Node 8 is the oldest version in LTS. I suggest we only officially support node 8 and higher from Commander v4. 

However, I suggest we actually target node 6 when considering language features and not deliberately break node 6 until there is good reason. That should cover most people using a current version of Commander while node 6 usage drops off. We no longer run tests on node 6 though so it won't be officially supported.

**Comment by shadowspawn** [comment]
I think the README addition is useful to be clear about what we are testing, but based on #1073 I think in practice we will still  need to bump major version if we deliberately make code changes to (say) require node 6 or 8. Without being noisy about the required version using `engines` we will still break people who might reasonably expect compatibility when updating within a major version.

 - https://docs.npmjs.com/files/package.json#engines

**Comment by shadowspawn** [comment]
I noticed today that that I used `class` for defining new `CommanderError` so we have indeed started requiring newer version of node, and not just a future consideration! Class was fully supported from node 6.

- https://node.green/#ES2015-functions-class
- https://docs.npmjs.com/files/package.json#engines

I'll do a PR adding engines field set to require node 6 to be explicit that Commander v4 onwards requires node 6 or higher.
